utils/zerovec/src/zerovec/mod.rs

```diff
@@ -777,15 +837,23 @@ where
     pub fn into_cow(self) -> Cow<'a, [T::ULE]> {
         if self.is_owned() {
             let vec = unsafe {
-                let len = (&*self.buf).len();
-                let ptr = self.buf as *mut T::ULE;
-                let capacity = self.capacity;
+                // If self is owned, then we know
+                // for a fact that it came from the parts of a Vec
+                // (via Self::new_owned())
+                let len = (&*self.vector.buf).len();
+                let ptr = self.vector.buf as *mut T::ULE;
+                let capacity = self.vector.capacity;
                 mem::forget(self);
                 Vec::from_raw_parts(ptr, len, capacity)
             };
             Cow::Owned(vec)
         } else {
-            let slice = unsafe { &*self.buf };
+            // We can extend the lifetime of the slice to 'a
+            // since we know it is borrowed
+            let slice = unsafe { self.vector.as_arbitrary_slice() };
+            // The borrowed destructor is a no-op, but we want to prevent
+            // the check being run
+            mem::forget(self);
```

Comment:
Nit: Shouldn't the compiler be able to figure this out on its own?

Reply:
Usually, yes, but it doesn't hurt to help the compiler here.